src/overlay/DialogHeader.tsx

```diff
@@ -0,0 +1,42 @@
+import { Box } from "../../styled-system/jsx";
+import { Heading } from "../typography/Heading";
+import { Paragraph } from "../typography/Paragraph";
+
+import type { FC, ReactNode } from "react";
+
+import type { BoxProps } from "../../styled-system/jsx";
+
+export interface DialogHeaderProps extends Omit<BoxProps, "title"> {
+  /**
+   * Title displayed as a heading in the dialog header.
+   */
+  title?: ReactNode | string;
+
+  /**
+   * Optional description displayed below the title.
+   */
+  description?: ReactNode | string;
+}
+
+/**
+ * Dialog header component with title and optional description.
+ */
+export const DialogHeader: FC<DialogHeaderProps> = ({ title, description, ...props }) => {
```

Comment:
Header? or Title?

If it has a description it should be another component like I mentioned, possible called DescriptiveTitle